pkg/postgres/query/builder_test.go

```diff
@@ -273,7 +280,187 @@ func TestQueryBuilder(t *testing.T) {
+				Filter: &filter.Request{
+					Fields: []filter.RequestField{
+						{
+							Name:     "severity",
+							Operator: filter.CompareOperatorIsGreaterThanOrEqualTo,
+							Value:    5.3,
+						},
+						{
+							Name:     "severity",
+							Operator: filter.CompareOperatorIsLessThanOrEqualTo,
+							Value:    8.2,
+						},
+					},
+					Operator: filter.LogicOperatorOr,
+				},
+				Paging: &paging.Request{
+					PageIndex: 2,
+					PageSize:  5,
+				},
+				Sorting: &sorting.Request{
+					SortColumn:    "started",
+					SortDirection: "desc",
+				},
+			},
+			wantQuery: `WHERE "severity_col_name" >= ? OR "severity_col_name" <= ? ORDER BY ? DESC OFFSET 2 LIMIT 5`,
+			wantArgs:  []any{5.3, 8.2, "started_col_name"},
+		},
+		{
+			name: "build query with filter paging, sorting, greater than && less than operators",
+			mockArg: query.ResultSelector{
+				Filter: &filter.Request{
+					Fields: []filter.RequestField{
+						{
+							Name:     "severity",
+							Operator: filter.CompareOperatorIsGreaterThanOrEqualTo,
+							Value:    []any{5.3, 4.3},
+						},
+						{
+							Name:     "severity",
+							Operator: filter.CompareOperatorIsLessThanOrEqualTo,
+							Value:    []any{8.2, 2.1},
+						},
+					},
+					Operator: filter.LogicOperatorOr,
+				},
+				Paging: &paging.Request{
+					PageIndex: 2,
+					PageSize:  5,
+				},
+				Sorting: &sorting.Request{
+					SortColumn:    "started",
+					SortDirection: "desc",
+				},
+			},
+			wantQuery: `WHERE "severity_col_name" >= GREATEST(?, ?) OR "severity_col_name" <= LEAST(?, ?) ORDER BY ? DESC OFFSET 2 LIMIT 5`,
+			wantArgs:  []any{5.3, 4.3, 8.2, 2.1, "started_col_name"},
+		},
+		{
+			name: "build query with filter paging, sorting, begins with && contains compare operators",
+			mockArg: query.ResultSelector{
+				Filter: &filter.Request{
+					Fields: []filter.RequestField{
+						{
+							Name:     "other_filter_field",
+							Operator: filter.CompareOperatorBeginsWith,
+							Value:    []any{"some text"},
+						},
+						{
+							Name:     "other_filter_field",
+							Operator: filter.CompareOperatorContains,
+							Value:    []any{"another text"},
+						},
+					},
+					Operator: filter.LogicOperatorOr,
+				},
+				Paging: &paging.Request{
+					PageIndex: 2,
+					PageSize:  5,
+				},
+				Sorting: &sorting.Request{
+					SortColumn:    "started",
+					SortDirection: "desc",
+				},
+			},
+			wantQuery: `WHERE ("other_filter_field_col_name" ILIKE ? || '%') OR ("other_filter_field_col_name" ILIKE '%' || ? || '%') ORDER BY ? DESC OFFSET 2 LIMIT 5`,
+			wantArgs:  []any{"some text", "another text", "started_col_name"},
+		},
+		{
+			name: "build query with filter paging, sorting, and date operators",
+			mockArg: query.ResultSelector{
+				Filter: &filter.Request{
+					Fields: []filter.RequestField{
+						{
+							Name:     "published",
+							Operator: filter.CompareOperatorIsGreaterThanOrEqualTo,
+							Value:    5.3,
```

Comment:
This seems to be a duplicate test case, but your points have been noted. 
- I changed the new test case to omit `sorting` and `paging` since that has already been covered in previous test cases. 

Checkout the new tests: [b91d841](https://github.com/greenbone/opensight-golang-libraries/pull/77/commits/b91d841c0c21998f5f1224dbd63e0c7a59896430)